test-case-reporting/app.ts

```diff
@@ -42,6 +42,15 @@ function renderTestRunList(testRuns: Array<TestRun>): string {
   return Mustache.render(testRunListTemplate, {testRuns});
 }
 
+let buildListTemplate = '';
+function renderBuildList(builds: Array<Build>): string {
+  if (!buildListTemplate) {
+    buildListTemplate = fs.readFileSync('./static/build-list.html').toString();
+  }
+
+  return Mustache.render(buildListTemplate, {builds});
+}
```

Comment:
```javascript
const templateCache = {};
function render(templateName, data) {
  if (!templateCache[templateName]) {
    templateCache[templateName] = fs.readFileSync(
      `${TEMPLATE_DIR}/${templateName}.html`
    ).toString();
  }

  return Mustache.render(templateCache[templateName], data)
}
...
render('build-list', {builds});
...
render('test-run-list', {testRuns});
```

Reply:
I like this approach, but how do we typecheck it? `renderBuildList` takes an array of Builds, while `renderTestRuns` takes an array of TestRuns. We _could_ make a type alias for Array<TestRuns> & Array<Builds> & whatever else we eventually want to render.

Reply:
Done putting that approach in, PTAL at how I (kind of) did the typechecking.